examples/json_consumer.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# This is a simple example of the SerializingProducer using JSON.
+#
+import argparse
+
+from confluent_kafka import DeserializingConsumer
+from confluent_kafka.schema_registry import SchemaRegistryClient
+from confluent_kafka.schema_registry.json_schema import JsonDeserializer
+from confluent_kafka.serialization import StringDeserializer
+
+
+class User(object):
+    """
+    User record
+
+    Args:
+        name (str): User's name
+
+        favorite_number (int): User's favorite number
+
+        favorite_color (str): User's favorite color
+
+    """
+    def __init__(self, name=None, favorite_number=None, favorite_color=None):
+        self.name = name
+        self.favorite_number = favorite_number
+        self.favorite_color = favorite_color
+
+
+def dict_to_user(ctx, obj):
+    """
+    Converts object literal(dict) to a User instance.
+
+    Args:
+        ctx (SerializationContext): Metadata pertaining to the serialization
+            operation.
+
+        obj (dict): Object literal(dict)
+
+    """
+    if obj is None:
+        return None
+
+    return User(name=obj['name'],
+                favorite_number=obj['favorite_number'],
+                favorite_color=obj['favorite_color'])
+
+
+def main(args):
+    topic = args.topic
+
+    schema_str = """
+    {
+      "$schema": "http://json-schema.org/draft-07/schema#",
+      "title": "User",
+      "description": "A Confluent Kafka Python User",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "User's name",
+          "type": "string"
+        },
+        "favorite_number": {
+          "description": "User's favorite number",
+          "type": "number",
+          "exclusiveMinimum": 0
+        },
+        "favorite_color": {
+          "description": "User's favorite color",
+          "type": "string"
+        }
+      },
+      "required": [ "name", "favorite_number", "favorite_color" ]
+    }
+    """
+
+    sr_conf = {'url': args.schema_registry}
+    schema_registry_client = SchemaRegistryClient(sr_conf)
```

Comment:
given it's not used, i don't think you should construct it. misleading example.

Reply:
good point

Reply:
It's now optional but this gives us an asymmetric API (which also differs from Avro's). We could also just remove it and make it optional in a future release if we deem it necessary. We'd just need to makes sure we are comfortable always having to treat it as if the SR may not be there which might be fine.